### DISCUSSION TITLE
Tailwind Cli generating invalid CSS comments?

### DISCUSSION THREAD

**Discussion by TutorialDoctor** [tailwindlabs/tailwindcss#6808] [Help]
I'm using `tailwindcss": "^3.0.7` and the following build script in my package.json:

`build-css": "tailwindcss build app/assets/stylesheets/tailwind.scss -o app/assets/stylesheets/tailwind.css`

Which generates a file that has the following lines:

```
*, ::before, ::after {
  --tw-translate-x: 0;
  --tw-translate-y: 0;
  --tw-rotate: 0;
  --tw-skew-x: 0;
  --tw-skew-y: 0;
  --tw-scale-x: 1;
  --tw-scale-y: 1;
  --tw-transform: translateX(var(--tw-translate-x)) translateY(var(--tw-translate-y)) rotate(var(--tw-rotate)) skewX(var(--tw-skew-x)) skewY(var(--tw-skew-y)) scaleX(var(--tw-scale-x)) scaleY(var(--tw-scale-y));
  --tw-blur: var(--tw-empty,/*!*/ /*!*/);
  --tw-brightness: var(--tw-empty,/*!*/ /*!*/);
  --tw-contrast: var(--tw-empty,/*!*/ /*!*/);
  --tw-grayscale: var(--tw-empty,/*!*/ /*!*/);
  --tw-hue-rotate: var(--tw-empty,/*!*/ /*!*/);
  --tw-invert: var(--tw-empty,/*!*/ /*!*/);
  --tw-saturate: var(--tw-empty,/*!*/ /*!*/);
  --tw-sepia: var(--tw-empty,/*!*/ /*!*/);
  --tw-drop-shadow: var(--tw-empty,/*!*/ /*!*/);
  --tw-filter: var(--tw-blur) var(--tw-brightness) var(--tw-contrast) var(--tw-grayscale) var(--tw-hue-rotate) var(--tw-invert) var(--tw-saturate) var(--tw-sepia) var(--tw-drop-shadow);
}
```

I'm using Rails 5 and the comments here cause my specs to fail. with a Sass Syntax error.

`Sass::SyntaxError: Invalid CSS after "..pty,/*!*/ /*!/": expected function argument, was ");"`

For the longest I thought additional existing files were getting purged or something. What purpose do these lines serve and if they don't serve a purpose how can I remove them from being generated?

**Comment by runderscorer** [comment]
Did you ever figure this out? I’m also using Rails 5 and it generates a CSS file with some properties that are missing values.

**Comment by TutorialDoctor** [comment]
Our workaround was to not compress CSS for the RSPEC tests and to move the generated tailwind.css file into the public directory instead of the assets directory. This, however doesn't resolve the issue, it's just what we did to meet the timeline we have.

**Comment by adamwathan** [comment]
Hey! The CSS you shared there is 100% valid, but the error makes it look like some other process is removing the closing parentheses on one of the variables somehow. Can you try disabling Sass (assuming you’re not actually using it)?
